app/presenters/collection_presenter.rb

```diff
@@ -0,0 +1,24 @@
+class CollectionPresenter
+  include Enumerable
+
+  attr_reader :collection
+  private :collection
+
+  cattr_accessor :presenter
```

Comment:
@RamilGilmanov `CollectionPresenter` should know how what presenter use for it's items.
Basically that was wrong solution, checkout updated version